### PR TITLE
Make the generate_homepage script more generic. 

### PR DESCRIPTION
**Pull request by whereistanya**
Generate a page from a yaml config.

### CODE REVIEW

File: generate_homepage_from_content.py [hunk 1]
```diff
@@ -147,11 +150,15 @@ def html_element_from_content(content_dict):
...
-    return list(csv.DictReader(open(filepath)))
-
+def content_from_yaml(filepath):
+    """Given the filepath of a yaml file containing content, returns a dict of
+    content information."""
+    with open(filepath) as stream:
+      try:
+        contents = yaml.load(stream)
+      except yaml.YAMLError as ex:
+        print ex
```
Comment (maiamcc) [inline generate_homepage_from_content.py]: I don't see much point to catching this error if we don't add any additional info to it OR break the func. I would think we would WANT to `raise` the exception rather just printing the exception and potentially returning nonsense--which will either lead to more, weirder errors down the line, or us generating a blank content page and possibly overwriting a file.

I'd recommend just not catching the error--if something goes wrong, it'll raise an exception and crash the program--or if you are gonna catch it, just do so to add additional info (human readable string like "Something went wrong trying to read content from the yaml file!") and `raise` the exception anyway.

File: generate_homepage_from_content.py [hunk 2]
```diff
@@ -27,7 +29,8 @@
 # HTML templates
 CONTENT_CONTAINER = '<div class="content %s">%s</div>'
 WEBSITE_CONTENT_TEMPLATE = ('<span class="website-content">%s</span>\n'
-                            '<span class="website-url">%s</span>')
+                            '<span class="website-url">'
+                            '<a href="%s">%s</a></span><br><br>')
```
Comment (maiamcc) [inline generate_homepage_from_content.py]: This `span` element may have been changed to `div` in @jillh510's latest diff, so maybe a moot point...? But regardless, I'd say, don't use `<br>` and accomplish the spacing with css styling instead.

**Review by maiamcc — APPROVED** [review]
Awesome stuff! Left a few small comments but when you fix those, you're good to merge as far as I'm concerned! (Unless @jillh510 also wants to take a look?)

File: generate_homepage_from_content.py (hunk 1 above)
Comment (whereistanya) [reply generate_homepage_from_content.py]: Good point. Removed the try/except entirely.

File: generate_homepage_from_content.py (hunk 2 above)
Comment (whereistanya) [reply generate_homepage_from_content.py]: I'll take out the line breaks and wait for Jill's change. And actually, I see she linkifies the url as well, so I'll remove that from mine to make a cleaner merge.

**Pull request merged**